### PR TITLE
update: prevent re-download on every update

### LINKED ISSUE #8751: update: import-url updates will get re-downloaded even if nothing changes
# Bug Report

## Description

This is a regression starting with https://github.com/iterative/dvc/pull/8024. After `dvc import-url`, `dvc update` will always re-download data even if it hasn't changed.

### Reproduce

For any external data, do `dvc import-url` followed by `dvc update`.

### Expected

Before this change, `dvc update` would return fast with output like `'wiki.tar.gz.dvc' didn't change, skipping`.

### PR DESCRIPTION
**Pull request by dtrifiro**
fixes #8751

### CODE REVIEW

File: tests/func/test_update.py [hunk 1]
```diff
@@ -176,17 +190,19 @@ def test_update_import_url(tmp_dir, dvc, workspace):
     assert dst.read_text() == "updated file content"
 
 
-def test_update_import_url_no_download(tmp_dir, dvc, workspace):
+@pytest.mark.parametrize("outs_exist", (False, True))
+def test_update_import_url_no_download(
```
Comment (dberenbaum) [inline tests/func/test_update.py:194]: Should there also be a `test_update_import_no_download` (`import` instead of `import-url`)?

**Comment by dberenbaum** [comment]
Seems like the behavior between `--no-download` and `--to-remote` is inconsistent. We can fix in this PR or follow up with another one. For `--to-remote`, the `outs` metadata is updated with the new info but the workspace remains untouched, while `--no-download` drop the `outs` metadata and deletes anything in the workspace.

File: tests/func/test_update.py (hunk 1 above)
Comment (dtrifiro) [reply tests/func/test_update.py:222]: Indeed. Added a test.

**Comment by dtrifiro** [comment]
Regarding `--to-remote` and `--no-download`, I think we can follow up later.

**Review by dberenbaum — APPROVED** [review]

**Pull request merged**